File: src/main/java/Javier.java
public class Javier {
    public static Object[] intersect(Object[] lizt1, Object[] lizt2, int index) {

        if (lizt1 == null || lizt1.length == 0) {
            return new Object[]{};
        }

        Object head = lizt1[index];

        boolean found = false;

        for (Object o : lizt2) {
            if (o == head || (o != null && o.equals(head))) {
                found = true;
                break;
            }
        }

        Object[] rest = intersect(lizt1, lizt2, index + 1);

        if (found) {
            Object[] result = new Object[rest.length + 1];
            result[0] = head;

            System.arraycopy(rest, 0, result, 1, rest.length);
            return result;
        }

        return rest;
    }
}

File: src/main/clojure/Javier.clj
(ns Javier)

(defn same [lizt1 lizt2]
  (cond
    ;; both lists empty at same time
    (empty? lizt1) (empty? lizt2)

    ;; only one empty → not equal
    (empty? lizt2) false

    ;; compare first elements
    (= (first lizt1) (first lizt2))
    (same (rest lizt1) (rest lizt2))

    ;; mismatch case
    :else false))

(defn intersect [lizt1 lizt2]
  (cond
    ;; base case
    (empty? lizt1) '()

    ;; if first element exists in lizt2
    (some #(= % (first lizt1)) lizt2)
    (cons (first lizt1)
          (intersect (rest lizt1) lizt2))

    ;; otherwise skip it
    :else
    (intersect (rest lizt1) lizt2)))